appeals/web/src/server/appeals/appeal-details/application-decision-date/application-decision-date.controller.js

```diff
@@ -0,0 +1,201 @@
+				'Application decision date removed'
+			);
+
+			delete request.session.applicationDecisionDate;
+
+			return response.redirect(
+				`/appeals-service/appeal-details/${currentAppeal.appealId}/appellant-case`
+			);
+		}
+	} catch (error) {
+		logger.error(error);
+	}
+
+	return response.status(500).render('app/500.njk');
+};
+
+/**
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+export const getChangeApplicationSetDecisionDate = async (request, response) => {
+	return renderChangeApplicationSetDecisionDate(request, response);
+};
+
+/**
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+const renderChangeApplicationSetDecisionDate = async (request, response) => {
+	try {
+		const { errors, currentAppeal, apiClient } = request;
+		const appellantCaseData = await getAppellantCaseFromAppealId(
+			apiClient,
+			currentAppeal.appealId,
+			currentAppeal.appellantCaseId
+		);
+
+		const mappedPageContents = changeApplicationSetDecisionDatePage(
+			currentAppeal,
+			appellantCaseData,
+			request.session.applicationDecisionDate
+		);
+
+		return response
+			.status(200)
+			.render('patterns/change-page.pattern.njk', { pageContent: mappedPageContents, errors });
+	} catch (error) {
+		logger.error(error);
+		delete request.session.applicationDecisionDate;
+		if (error instanceof HTTPError && error.response.statusCode === 404) {
+			return response.status(404).render('app/404.njk');
+		} else {
+			return response.status(500).render('app/500.njk');
+		}
+	}
+};
+
+/**
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+export const postChangeApplicationSetDecisionDate = async (request, response) => {
+	console.log(request.body);
+	request.session.applicationDecisionDate = {
+		day: request.body['application-decision-date-day'],
+		month: request.body['application-decision-date-month'],
+		year: request.body['application-decision-date-year']
+	};
+
+	if (request.errors) {
+		return renderChangeApplicationSetDecisionDate(request, response);
+	}
+
+	try {
+		const { currentAppeal, apiClient, session } = request;
+
+		const { appealId, appellantCaseId } = currentAppeal;
+
+		const updatedApplicationDecisionDay = parseInt(session.applicationDecisionDate.day, 10);
+		const updatedApplicationDecisionMonth = parseInt(session.applicationDecisionDate.month, 10);
+		const updatedApplicationDecisionYear = parseInt(session.applicationDecisionDate.year, 10);
+
+		if (
+			Number.isNaN(updatedApplicationDecisionDay) ||
+			Number.isNaN(updatedApplicationDecisionMonth) ||
+			Number.isNaN(updatedApplicationDecisionYear)
+		) {
+			return response.status(500).render('app/500.njk');
```

Comment:
I think 400 is appropriate here: Bad Request? Not sure if we have a page for it.

Reply:
There is no 400 page. I could create one if you think it's needed?

Reply:
No that's OK, one to maybe visit post MVP. Validation should handle this anyway, so actually 500 is perhaps appropriate given that its unexpected!